src/main/java/org/myrobotlab/service/ProgramAB.java

```diff
@@ -711,12 +711,25 @@ public Session startSession(String path, String userName, String botName, java.u
     }
 
     session = new Session(this, userName, botInfo);
-    sessions.put(getSessionKey(userName, botName), session);
+    String sessionKey = getSessionKey(userName, botName);
+    sessions.put(sessionKey, session);
 
     log.info("Started session for bot botName:{} , userName:{}", botName, userName);
     setCurrentSession(userName, botName);
+    
+    invoke("publishSession", sessionKey);

```

Comment:
we should probably check this.. and publish not just when a session is started, but when the session changes.

Reply:
:+1: